index.js

```diff
@@ -516,129 +518,59 @@ function Cardboard(config) {
-                    memo.push([bbox[0], bbox[1], bbox[2], -epsilon]);
-                    memo.push([bbox[0], epsilon, bbox[2], bbox[3]]);
-                    return memo;
-                }, []);
-            }
-        }
-
-        var tiles = bboxes.map(function(bbox) {
-            return tilebelt.bboxToTile(bbox);
-        });
-
-        // Deduplicate subquery tiles.
-        uniq(tiles, function(a, b) {
-            return !tilebelt.tilesEqual(a, b);
-        });
-
-        if (tiles.length > 1) {
-            // Filter out the z0 tile -- we'll always search it eventually.
-            tiles = _.filter(tiles, function(item) {
-                return item[2] !== 0;
-            });
-        }
-
-        tiles.forEach(function(tile) {
-            var tileKey = tilebelt.tileToQuadkey(tile);
-
-            // First find features indexed in children of this tile
-            var query = {
-                cell: { BEGINS_WITH: 'cell!' + tileKey },
-                dataset: { EQ: dataset }
-            };
-
-            var options = {
-                pages: 0,
-                index: 'cell',
-                filter: {
-                    west: { LE: bbox[2] },
-                    east: { GE: bbox[0] },
-                    north: { GE: bbox[1] },
-                    south: { LE: bbox[3] }
-                }
-            };
-            q.defer(config.dyno.query, query, options);
+        options = _.defaults(options, { limit: 100 });
 
-            // Travel up the parent tiles, finding features indexed in each
-            var parentTileKey = tileKey.slice(0, -1);
+        // First find features indexed in children of this tile
+        var query = {
+            dataset: { EQ: dataset },
+            id: {BEGINS_WITH: 'id!'}
+        };
 
-            while (tileKey.length > 0) {
-                query.cell = { EQ: 'cell!' + parentTileKey };
-                q.defer(config.dyno.query, query, options);
-                if (parentTileKey.length === 0) break;
-                parentTileKey = parentTileKey.slice(0, -1);
+        var queryOptions = {
+            pages: 1,
+            limit: options.limit,
+            filter: {
+                west: { LE: bbox[2] },
+                east: { GE: bbox[0] },
+                north: { GE: bbox[1] },
+                south: { LE: bbox[3] }
             }
-        });
-
-        q.awaitAll(function(err, items) {
-            if (err) return callback(err);
+        };
 
-            items = _.flatten(items);
+        if (options.start) {
+            queryOptions.start =  {
+                dataset: dataset,
+                id: 'id!'+options.start
+            };
+        }
 
-            // Reduce the response's records to the set of
-            // records with unique ids.
-            uniq(items, function(a, b) {
-                return a.id !== b.id;
-            });
+        var maxPages = 10;
+        var page = 0;
+        var combinedFeatures = [];
```

Comment:
We dont limit on size with the list queries right now (because the features could be on s3) but we limit to 1 page or 100 features.  I dont think we need this right now, and in the future this will be limited by dynamo if we keep all the features in there.